app/lib/scope_transformer.rb

```diff
@@ -2,15 +2,23 @@
 
 class ScopeTransformer < Parslet::Transform
   class Scope
+    PROFILE_TERM   = 'profile'
     DEFAULT_TERM   = 'all'
     DEFAULT_ACCESS = %w(read write).freeze
+    READ_ONLY_ACCESS = %w(read).freeze
 
     attr_reader :namespace, :term
 
     def initialize(scope)
       @namespace = scope[:namespace]&.to_s
-      @access    = scope[:access] ? [scope[:access].to_s] : DEFAULT_ACCESS.dup
       @term      = scope[:term]&.to_s || DEFAULT_TERM
+
+      # override for profile scope which is read only
+      @access = if @term == PROFILE_TERM
+                  READ_ONLY_ACCESS.dup
+                else
+                  scope[:access] ? [scope[:access].to_s] : DEFAULT_ACCESS.dup
+                end
```

Comment:
I *think* this may just be more readable as something like:

```ruby
# override for profile scope which is read-only
@access = %w(read) if @term == 'profile'
```

The overhead of doing the assignment twice is negligible, and I think that special logic being self-contained in a single line is more readable than involving multiple constants.

Reply:
Have done this.